app/common/directives/role-selector.directive.js

```diff
@@ -37,10 +37,12 @@ function RoleSelectorController($scope, RoleEndpoint, $translate, _) {
         if (_.indexOf($scope.model.role, admin.name) === -1) {
             $scope.model.role.push(admin.name);
         }
+        document.getElementById('add_roles').click();
     }
 
     // adding all available roles to model if user clicks 'Everyone'
     function setEveryone() {
         $scope.model.role = null;
+        document.getElementById('add_everyone').click();
```

Comment:
Instead of doing the click here and on row 40, we need mimic the behaviour that is going on in the input-elements, otherwise this function will be run twice. What is happening in the input-elements is that the model used, "$scope.everyone", is set to either true or false. This makes the dropdown with the roles visible or hidden (see this line: https://github.com/ushahidi/platform-client/blob/develop/app/common/directives/role-selector.html#L49). So in the setEveryone-function, we need to set $scope.everyone = true and in the setAdmin-function, we need to set it to false.

Reply:
Alright, I will fix this.